gcli/src/app.rs

```diff
@@ -90,6 +90,16 @@ pub trait App: Parser + Sync {
     /// Exec program from the parsed arguments.
     async fn exec(&self) -> anyhow::Result<()>;
 
+    /// Get gear api without signing in with password.
+    async fn api(&self) -> anyhow::Result<GearApi> {
+        let endpoint = self.endpoint().clone();
+        let timeout = self.timeout();
+        Api::new_with_timeout(endpoint.as_deref(), Some(timeout))
+            .await
+            .map(Into::into)
```

Comment:
If I understand correctly, then `.map(Into::into)` here is Alice’s account. Maybe we should use the primary address from gring?

Reply:
Makes sense, this is currently a workaround for adapting that `gclient` force us using a signer but not just a storage api, so the signer discover logic here should never failed

Oh I forgot `gcli info balance` should print the signer's balance

Reply:
The problem here is that we have to input `password` on `gcli info balance`, need to refactor on this kind of needs that users can run commands without inputing the `password`

Reply:
I'm thinking of introducing a method reading the address of the primary account from `gring`

Reply:
f606e20